python/kserve/kserve/model.py

```diff
@@ -75,19 +89,43 @@ def __init__(self, name: str):
         self._http_client_instance = None
         self._grpc_client_stub = None
 
-    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR):
-        request = await self.preprocess(body) if inspect.iscoroutinefunction(self.preprocess) \
-            else self.preprocess(body)
+    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR, headers={}):
+        request_id = headers.get("X-RequestID")
+
+        # latency vars
+        preprocess_ms = 0
+        explain_ms = 0
+        predict_ms = 0
+        postprocess_ms = 0
+
+        with PRE_HIST_TIME.time():
+            start = time.time()
+            request = await self.preprocess(body) if inspect.iscoroutinefunction(self.preprocess) \
+                else self.preprocess(body)
+            preprocess_ms = get_latency_ms(start, time.time())
         request = self.validate(request)
         if model_type == ModelType.EXPLAINER:
-            response = (await self.explain(request)) if inspect.iscoroutinefunction(self.explain) \
-                else self.explain(request)
+            with EXPLAIN_HIST_TIME.time():
+                start = time.time()
+                response = (await self.explain(request)) if inspect.iscoroutinefunction(self.explain) \
+                    else self.explain(request)
+                explain_ms = get_latency_ms(start, time.time())
         elif model_type == ModelType.PREDICTOR:
-            response = (await self.predict(request)) if inspect.iscoroutinefunction(self.predict) \
-                else self.predict(request)
+            with PREDICT_HIST_TIME.time():
+                start = time.time()
+                response = (await self.predict(request)) if inspect.iscoroutinefunction(self.predict) \
+                    else self.predict(request)
+                predict_ms = get_latency_ms(start, time.time())
         else:
             raise NotImplementedError
-        response = self.postprocess(response)
+
+        with POST_HIST_TIME.time():
+            start = time.time()
+            response = self.postprocess(response)
+            postprocess_ms = get_latency_ms(start, time.time())
+
+        logging.info(f"requestId: {request_id}, preprocess_ms: {preprocess_ms}, explain_ms: {explain_ms}, predict_ms: {predict_ms}, postprocess_ms: {postprocess_ms}")
+
```

Comment:
note on the code: calling `start = time.time()` seems redundant here - Ive put some thought into it but not sure if the alternatives are cleaner. afaik, using a decorator would mean we have to log each step's time, so instead of one single log at the end of __call__ we would have one for each step, probably not great to expand the logs that much. 

im open to suggestions!